pkg/filesystem/path/builder.go

```diff
@@ -66,11 +83,44 @@ func (b *Builder) GetUNIXString() string {
 	return out.String()
 }
 
+// GetWindowsString returns a string representation of the path for use on
+// UNIX-like operating systems. We do not create a windows-style path.
+func (b *Builder) GetWindowsString() (string, error) {
+	// Emit pathname components.
+	prefix := ""
+	if b.absolute {
+		prefix = "\\"
+	}
+
+	var out strings.Builder
+	if b.driveLetter != rune(0) {
+		out.WriteString(string(b.driveLetter) + ":")
+	}
+
+	for _, component := range b.components {
+		out.WriteString(prefix)
+		out.WriteString(component)
```

Comment:
This does not take into account whether the component is valid on Windows. Could we please extend it to check for control characters, backslashes, colons, etc. etc. etc.?

Reply:
:+1: Do we want them in-line here? And should I create test-cases for them through the Parse code path, or should I create a component-verification function that can be tested directly?